crypto/src/bls.rs

```diff
@@ -161,33 +155,47 @@ fn prepend_public_key(msg: &[u8], pk: &PublicKeyBls) -> Vec<u8> {
     message_with_pk
 }
 
+/// Generate a secret key from initial key material.
+pub fn key_from_ikm(
+    ikm: [u8; 32],
+) -> Result<(SecretKeyBls, PublicKeyBls, ContractTz4Hash), CryptoError> {
```

Comment:
For now, shall we leave it? 
If used in a kernel, it would only ever be tests, as the kernel would never use the `secret key` etc directly?

Exactly - can always do `(sk, pk, _) = bls::key_from_ikm(ikm)?;`

Reply:
true, makes total sense.

Reply:
have updated 👍🏻